### PR TITLE
BGE settings for Reblocking and JointCalling pipelines

### PR DESCRIPTION
**Pull request by meganshand**
### Description

This is a draft of how we'd like to run joint calling on BGE with a few changes to the current pipeline:

- Use an updated variant calling interval list (bge v1.1 calling intervals)
- Reblocking
  * Use an updated version of ReblockGVCFs to work with DRAGEN input
  * Added argument to strip “PRI” annotation: `--format-annotations-to-remove PRI`
  * Add argument to move site level filters to genotype level instead: `--add-site-filters-to-genotype` 
  * No need to do extra blocking at this step since the block sizes are now set in the DRAGEN/single sample pipeline
- Filtering
  * Allow for the option to use the newer VETS pipeline rather than VQSR in JointGenotyping
  * Hard filter sites that land outside of the targeted regions (but within the exome variant calling interval list)
  * Train the VETS model only over the targeted regions of the BGE (where there's expected coverage)
  * Provide the SCORE annotation calculated by the VETS model for all variants, but don't use it as an applied filter so that users can choose their own cutoff values

Making two paths in the JointCalling pipeline for two different filtering strategies was a bit messy, so I'm happy to hear if there are any recommendations on how to clean that up. I also tried to run the tests that already exist for these workflows, but ran into errors that seemed unrelated. 

@jessicaway For now I left out any changes to the ByChromosome workflows and we can cross that bridge when we come to it.

----

### Checklist 

If you can answer "yes" to the following items, please add a checkmark next to the appropriate checklist item(s) and notify our WARP documentation team by tagging either @ekiernan or @kayleemathews in a comment on this PR.

- [x] Did you add inputs, outputs, or tasks to a workflow?
- [x] Did you modify, delete or move: file paths, file names, input names, output names, or task names?
- [x] If you made a changelog update, did you update the pipeline version number?

### CODE REVIEW

**Comment by meganshand** [comment]
@jessicaway @samuelklee This is ready for review. As discussed in the meeting this morning the two remaining issues are:

- We'll need to update the version of GATK once it's been released
- Test files still need to be copied into the appropriate bucket for all the tests to pass here

I don't think either of these should stop an initial review though. Thanks!

**Comment by meganshand** [comment]
@ekiernan This PR will need some documentation updates for the BGE pipeline. I also submitted a request through the portal. Happy to work with you on this.

**Comment by samuelklee** [comment]
Great, thanks @meganshand! Should be able to get an initial review back to you in the next couple days.

Would be great to also get https://github.com/broadinstitute/gatk/pull/8131 in before the next GATK release. Let's chat about whether that necessitates any changes here---I don't think it should, but that PR has been on the back burner for a bit, so I'll have to double check.

File: pipelines/broad/dna_seq/germline/joint_genotyping/JointGenotyping.wdl [hunk 1]
```diff
@@ -192,58 +196,59 @@ workflow JointGenotyping {
...
-      use_allele_specific_annotations = allele_specific_annotations,
-      disk_size_gb = small_disk
-  }
-
-  if (num_gvcfs > snps_variant_recalibration_threshold) {
-    call Tasks.SNPsVariantRecalibratorCreateModel {
+  
+  if (run_vets) {
+    String allele_specific_extra_args = if allele_specific_annotations then " --use-allele-specific-annotations " else ""
+    String resource_args = " --resource:hapmap,training=true,calibration=true " + hapmap_resource_vcf +
```
Comment (samuelklee) [inline pipelines/broad/dna_seq/germline/joint_genotyping/JointGenotyping.wdl:202]: Would be nice to expose this and demote the default resources from required arguments at some point, but this is fine for now.

File: pipelines/broad/dna_seq/germline/joint_genotyping/JointGenotyping.wdl [hunk 2]
```diff
@@ -192,58 +196,59 @@ workflow JointGenotyping {
...
-      dbsnp_resource_vcf = dbsnp_resource_vcf,
-      dbsnp_resource_vcf_index = dbsnp_resource_vcf_index,
-      use_allele_specific_annotations = allele_specific_annotations,
-      disk_size_gb = small_disk
-  }
-
-  if (num_gvcfs > snps_variant_recalibration_threshold) {
-    call Tasks.SNPsVariantRecalibratorCreateModel {
+  
+  if (run_vets) {
```
Comment (samuelklee) [inline pipelines/broad/dna_seq/germline/joint_genotyping/JointGenotyping.wdl:200]: Ideally, we'd have separate VQSR and VETS subworkflows, and the latter would be a direct import of the JointVcfFiltering WDL in the GATK repo. Either we'll get there in the future or we'll remove VQSR, let's see which happens first!

File: pipelines/broad/dna_seq/germline/joint_genotyping/reblocking/ReblockGVCF.changelog.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# 2.1.6
```
Comment (samuelklee) [inline pipelines/broad/dna_seq/germline/joint_genotyping/reblocking/ReblockGVCF.changelog.md:1]: Do you need to update some of these changelogs now that #1081 is in?

File: pipelines/broad/dna_seq/germline/joint_genotyping/test_inputs/Scientific/bge.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  "JointGenotyping.hapmap_resource_vcf_index": "gs://gcp-public-data--broad-references/hg38/v0/hapmap_3.3.hg38.vcf.gz.tbi",
+  "JointGenotyping.mills_resource_vcf": "gs://gcp-public-data--broad-references/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz",
+  "JointGenotyping.mills_resource_vcf_index": "gs://gcp-public-data--broad-references/hg38/v0/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi",
+  "JointGenotyping.omni_resource_vcf": "gs://gcp-public-data--broad-references/hg38/v0/1000G_omni2.5.hg38.vcf.gz",
+  "JointGenotyping.omni_resource_vcf_index": "gs://gcp-public-data--broad-references/hg38/v0/1000G_omni2.5.hg38.vcf.gz.tbi",
+  "JointGenotyping.one_thousand_genomes_resource_vcf": "gs://gcp-public-data--broad-references/hg38/v0/1000G_phase1.snps.high_confidence.hg38.vcf.gz",
+  "JointGenotyping.one_thousand_genomes_resource_vcf_index": "gs://gcp-public-data--broad-references/hg38/v0/1000G_phase1.snps.high_confidence.hg38.vcf.gz.tbi",
+
+  "JointGenotyping.snp_recalibration_annotation_values": ["QD", "MQRankSum", "ReadPosRankSum", "FS", "MQ", "SOR", "DP"],
```
Comment (samuelklee) [inline pipelines/broad/dna_seq/germline/joint_genotyping/test_inputs/Scientific/bge.json:27]: Why are these not allele-specific annotations, when `use_allele_specific_annotations = true` below?

File: pipelines/broad/dna_seq/germline/joint_genotyping/test_inputs/Scientific/bge.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
+  "JointGenotyping.sample_name_map": "FILL_IN",
+  "JointGenotyping.callset_name": "bge_joint_genotyping_plumbing",
+  "JointGenotyping.run_vets": true,
+
+  "JointGenotyping.unpadded_intervals_file": "",
```
Comment (samuelklee) [inline pipelines/broad/dna_seq/germline/joint_genotyping/test_inputs/Scientific/bge.json:6]: Were `unpadded_intervals_file`, `eval_interval_list`, and `targets_interval_list` intentionally left empty for now?

File: tasks/broad/JointGenotypingTasks.wdl [hunk 6]
```diff
@@ -273,6 +273,7 @@ task HardFilterAndMakeSitesOnlyVcf {
 
     String variant_filtered_vcf_filename
     String sites_only_vcf_filename
+    File? targets_interval_list # filters out all variants outside of the targets interval list for targetted sequencing
```
Comment (samuelklee) [inline tasks/broad/JointGenotypingTasks.wdl:276]: `targetted` -> `targeted`.

I'm also unclear about whether such changes to tasks need to be reflected in a changelog somewhere?

**Review by samuelklee — CHANGES_REQUESTED** [review]
Thanks @meganshand, just a few minor comments/questions from me!

**Comment by meganshand** [comment]
The failing smart tests here are due to a change in the VCF header when updating to GATK 4.5.0.0:

```
< ##INFO=<ID=RAW_GT_COUNT,Number=3,Type=Integer,Description="Counts of genotypes w.r.t. the reference allele in the following order: 0/0, 0/*, */*, i.e. all alts lumped together; for use in calculating excess heterozygosity">
---
> ##INFO=<ID=RAW_GT_COUNT,Number=3,Type=Integer,Description="Counts of genotypes w.r.t. the reference allele: 0/0, 0/*, */*, i.e. all alts lumped together; for use in calculating excess heterozygosity">
```

This change seems harmless to me, but I'm not sure who needs to sign off on this change. The PR with the rest of the GATK updates to 4.5.0.0 is still in progress, but should be coming soon.

**Comment by meganshand** [comment]
Failing Scientific Tests that can be updated:

- JointGenotyping: BGE test is failing because the truth data is out of data. This can safely be updated as these differences are expected.
- ReblockGVCFs: Tests are failing because documentation on a header line changed (see comment above). This can safely be updated.
- VariantCalling: Same header line change as ReblockGVCFs. This can be updated.

**Review by jessicaway — APPROVED** [review]

**Review by nikellepetrillo — APPROVED** [review]

**Comment by nikellepetrillo** [comment]
after merging in the latest develop, we re-ran JG, variant calling, and reblocking against Megan's truth (which was previously validated):

https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/65222/console
https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/65156/console/
https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/65155/console

**Pull request merged**